atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/transaction/impl/SnapshotTransaction.java

```diff
@@ -749,11 +749,15 @@ private void validatePreCommitRequirementsOnReadIfNecessary(TableReference table
     }
 
     private boolean isValidationNecessaryOnReads(TableReference tableRef) {
-        return validateLocksOnReads && isThoroughlySwept(tableRef);
+        return validateLocksOnReads && requiresImmutableTimestampLocking(tableRef);
     }
 
     private boolean isValidationNecessaryOnCommit(TableReference tableRef) {
-        return !validateLocksOnReads && isThoroughlySwept(tableRef);
+        return !validateLocksOnReads && requiresImmutableTimestampLocking(tableRef);
+    }
+
+    private boolean requiresImmutableTimestampLocking(TableReference tableRef) {
+        return isThoroughlySwept(tableRef) || transactionConfig.get().lockImmutableTsOnReadOnlyTransactions();
```

Comment:
This means that the value in the config can change between when we start the transaction and any particular read (or between reads). Since this isn't a correctness issue, I don't think we should worry about it.

Note that this might have caused failures or precondition checks, depending on how our code was written. If the config changes to cause us to check immutable timestamp lock validity but it turns out we never took one (since the flag was false when we opened the transaction), then we might have failed if we had assumed that validation necessary meant that the immutable timestamp lock must exist.

Reply:
yeah, I was also concerned about this value changing in the middle of a transaction; but our check logic throws only if immutableTsLock exists and not valid anymore. If it doesn't exist in the first place, than it doesn't complain.